[question] Installing an executable with its dependencies through CMake with the new CMakeConfigDeps

### What is your question?

Hello !

Although I am well aware deployers exist to install components coming from another conan package, I would like to able to do so from a pure CMake way

In fact, I try to leverage ctest in order to run cmake install and copy the desired components (executables) and their dependencies to the required location, before running the tests

I am trying with the new `CMakeConfigDeps` to be able to reach the executables through the targets, but that does not seem to be enough:

```CMake
...
find_package(MyTarget)

get_target_property(WRAPPERGEN_EXEC MyTarget::wrappergenerator LOCATION) # The var is properly populated

add_executable(wrappergenerator IMPORTED GLOBAL)

set_property(TARGET wrappergenerator PROPERTY
             IMPORTED_LOCATION "${WRAPPERGEN_EXEC}")

install(RUNTIME_DEPENDENCY_SET my_app_deps
	PRE_EXCLUDE_REGEXES
		[=[api-ms-]=] # VC Redistibutable DLLs
		[=[ext-ms-]=] # Windows extension DLLs
		# [=[python]=]
		[[kernel32\.dll]] # Default in conan doc
		[[libc\.so\..*]] [[libgcc_s\.so\..*]] [[libm\.so\..*]] [[libstdc\+\+\.so\..*]] # Default in conan doc
	POST_EXCLUDE_REGEXES
		[=[.*system32\/.*\.dll]=] # Windows system DLLs
		[=[^\/(lib*|usr\/lib*|usr\/local\/lib*\/lib64)]=] # Unix system libraries
	DIRECTORIES
		${CONAN_RUNTIME_LIB_DIRS}

install(TARGETS wrappergenerator
	RUNTIME_DEPENDENCY_SET my_app_deps
	RUNTIME
		DESTINATION ${CMAKE_INSTALL_BINDIR}
	LIBRARY
		DESTINATION ${CMAKE_INSTALL_LIBDIR}
)
```

While I could simply run `install(FILES ${WRAPPERGEN_EXEC} ...`, that would not allow accessing the deps of the executable, would it?

In the end, would it be possible to add some more documentation regarding these scenarios? Whether it concerns executables or libraries? Thank you very much for your work.

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

Should've seen that simple mistake. Thank you for fixing it.

Indeed, it seems to work this way, and all the transitive libs install properly! Super nice!! 

However, if I set `app` as editable, the executable cannot seem to be found:

### Logs

<details><summary>Click to expand log</summary>

```cmd
(venv) C:\Users\Todiq\AppData\Local\Temp>conan editable add app
Reference 'app/1.0' in editable mode

(venv) C:\Users\Todiq\AppData\Local\Temp>conan build app -o "*:shared=True"

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows
[options]
*:shared=True

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows


======== Computing dependency graph ========
Graph root
    conanfile.py (app/1.0): C:\Users\Todiq\AppData\Local\Temp\app\conanfile.py
Requirements
    liba/1.0#0fbf481ef3fd42eb7fe3a14079aa6d3a - Cache
    libb/1.0#63571d919743c4cbc1e293cdbd8c3b01 - Cache

======== Computing necessary packages ========
Requirements
    liba/1.0#0fbf481ef3fd42eb7fe3a14079aa6d3a:a818caff19c116dafdc2337a9b30675bd75e8985#94f4c3445e9d525418a9d88a388c6c31 - Cache
    libb/1.0#63571d919743c4cbc1e293cdbd8c3b01:5527a2ac492ae26512f43737725dfdbdee4016aa#fe0fb4458131fab337a38446e5cb8e6b - Cache

======== Installing packages ========
liba/1.0: Already installed! (1 of 2)
libb/1.0: Already installed! (2 of 2)

======== Finalizing install (deploy, generators) ========
conanfile.py (app/1.0): Writing generators to C:\Users\Todiq\AppData\Local\Temp\app\build\generators
conanfile.py (app/1.0): Generator 'CMakeDeps' calling 'generate()'
conanfile.py (app/1.0): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(libb)
    target_link_libraries(... libb::libb)
conanfile.py (app/1.0): Generator 'CMakeToolchain' calling 'generate()'
conanfile.py (app/1.0): CMakeToolchain generated: conan_toolchain.cmake
conanfile.py (app/1.0): CMakeToolchain: Preset 'conan-default' added to CMakePresets.json.
    (cmake>=3.23) cmake --preset conan-default
    (cmake<3.23) cmake <path> -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE=generators\conan_toolchain.cmake  -DCMAKE_POLICY_DEFAULT_CMP0091=NEW
conanfile.py (app/1.0): CMakeToolchain generated: C:\Users\Todiq\AppData\Local\Temp\app\build\generators\CMakePresets.json
conanfile.py (app/1.0): CMakeToolchain generated: C:\Users\Todiq\AppData\Local\Temp\app\CMakeUserPresets.json
conanfile.py (app/1.0): Generating aggregated env files
conanfile.py (app/1.0): Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']

======== Calling build() ========
conanfile.py (app/1.0): Calling build()
conanfile.py (app/1.0): Running CMake.configure()
conanfile.py (app/1.0): RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/Todiq/AppData/Local/Temp/app" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:/Users/Todiq/AppData/Local/Temp/app"
-- Using Conan toolchain: C:/Users/Todiq/AppData/Local/Temp/app/build/generators/conan_toolchain.cmake
-- Conan toolchain: CMAKE_GENERATOR_TOOLSET=v143
-- Conan toolchain: Setting CMAKE_MSVC_RUNTIME_LIBRARY=$<$<CONFIG:Release>:MultiThreadedDLL>
-- Conan toolchain: C++ Standard 14 with extensions OFF
-- The CXX compiler identification is MSVC 19.44.35207.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.44.35207/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Target declared 'libb::libb'
-- Conan: Target declared 'liba::liba'
-- Configuring done (5.4s)
-- Generating done (0.0s)
-- Build files have been written to: C:/Users/Todiq/AppData/Local/Temp/app/build

conanfile.py (app/1.0): Running CMake.build()
conanfile.py (app/1.0): RUN: cmake --build "C:\Users\Todiq\AppData\Local\Temp\app\build" --config Release
Version MSBuild 17.14.8+a7a4d5af0 pour .NET Framework

C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(548,5): warning MSB8
029: Le répertoire intermédiaire ou le répertoire de sortie ne peut pas se trouver sous le répertoire temporaire car cela risque de
créer des problèmes avec la génération incrémentielle. [C:\Users\Todiq\AppData\Local\Temp\app\build\ZERO_CHECK.vcxproj]
  1>Checking Build System
C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(548,5): warning MSB8
029: Le répertoire intermédiaire ou le répertoire de sortie ne peut pas se trouver sous le répertoire temporaire car cela risque de
créer des problèmes avec la génération incrémentielle. [C:\Users\Todiq\AppData\Local\Temp\app\build\app.vcxproj]
  Building Custom Rule C:/Users/Todiq/AppData/Local/Temp/app/CMakeLists.txt
  app.cpp
  main.cpp
     Création de la bibliothèque C:/Users/Todiq/AppData/Local/Temp/app/build/Release/app.lib et de l'objet C:/Users/Todiq/AppD
  ata/Local/Temp/app/build/Release/app.exp
  app.vcxproj -> C:\Users\Todiq\AppData\Local\Temp\app\build\Release\app.exe
C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(548,5): warning MSB8
029: Le répertoire intermédiaire ou le répertoire de sortie ne peut pas se trouver sous le répertoire temporaire car cela risque de
créer des problèmes avec la génération incrémentielle. [C:\Users\Todiq\AppData\Local\Temp\app\build\ALL_BUILD.vcxproj]
  Building Custom Rule C:/Users/Todiq/AppData/Local/Temp/app/CMakeLists.txt


(venv) C:\Users\Todiq\AppData\Local\Temp>conan build . -o "*:shared=True" -c "tools.cmake.cmakedeps:new=will_break_next"

======== Input profiles ========
Profile host:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows
[options]
*:shared=True
[conf]
tools.cmake.cmakedeps:new=will_break_next

Profile build:
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Release
compiler.version=194
os=Windows


======== Computing dependency graph ========
Graph root
    conanfile.py: C:\Users\Todiq\AppData\Local\Temp\conanfile.py
Requirements
    app/1.0 - Editable
    liba/1.0#0fbf481ef3fd42eb7fe3a14079aa6d3a - Cache
    libb/1.0#63571d919743c4cbc1e293cdbd8c3b01 - Cache

======== Computing necessary packages ========
Requirements
    app/1.0:472cbe405284f0574862362b67fe72f21448e5b0 - Editable
    liba/1.0#0fbf481ef3fd42eb7fe3a14079aa6d3a:a818caff19c116dafdc2337a9b30675bd75e8985#94f4c3445e9d525418a9d88a388c6c31 - Cache
    libb/1.0#63571d919743c4cbc1e293cdbd8c3b01:5527a2ac492ae26512f43737725dfdbdee4016aa#fe0fb4458131fab337a38446e5cb8e6b - Cache

======== Installing packages ========
liba/1.0: Already installed! (1 of 3)
libb/1.0: Already installed! (2 of 3)
app/1.0: Rewriting files of editable package 'app' at 'C:\Users\Todiq\AppData\Local\Temp\app\build\generators'
app/1.0: Writing generators to C:\Users\Todiq\AppData\Local\Temp\app\build\generators
app/1.0: WARN: Using the new CMakeConfigDeps generator, behind the 'tools.cmake.cmakedeps:new' gate conf. This conf will change next release, breaking, so use it only for testing and dev
app/1.0: Generator 'CMakeDeps' calling 'generate()'
app/1.0: CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(libb)
    target_link_libraries(... libb::libb)
app/1.0: Generator 'CMakeToolchain' calling 'generate()'
app/1.0: CMakeToolchain generated: conan_toolchain.cmake
app/1.0: CMakeToolchain generated: C:\Users\Todiq\AppData\Local\Temp\app\build\generators\CMakePresets.json
app/1.0: CMakeToolchain generated: C:\Users\Todiq\AppData\Local\Temp\app\CMakeUserPresets.json
app/1.0: Generating aggregated env files
app/1.0: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']

======== Finalizing install (deploy, generators) ========
conanfile.py: Writing generators to C:\Users\Todiq\AppData\Local\Temp\build\generators
conanfile.py: WARN: Using the new CMakeConfigDeps generator, behind the 'tools.cmake.cmakedeps:new' gate conf. This conf will change next release, breaking, so use it only for testing and dev
conanfile.py: Generator 'CMakeDeps' calling 'generate()'
conanfile.py: CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(app)
conanfile.py: Generator 'CMakeToolchain' calling 'generate()'
conanfile.py: CMakeToolchain generated: conan_toolchain.cmake
conanfile.py: CMakeToolchain: Preset 'conan-default' added to CMakePresets.json.
    (cmake>=3.23) cmake --preset conan-default
    (cmake<3.23) cmake <path> -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE=generators\conan_toolchain.cmake  -DCMAKE_POLICY_DEFAULT_CMP0091=NEW
conanfile.py: CMakeToolchain generated: C:\Users\Todiq\AppData\Local\Temp\build\generators\CMakePresets.json
conanfile.py: CMakeToolchain generated: C:\Users\Todiq\AppData\Local\Temp\CMakeUserPresets.json
conanfile.py: Generating aggregated env files
conanfile.py: Generated aggregated env files: ['conanbuild.bat', 'conanrun.bat']

======== Calling build() ========
conanfile.py: Calling build()
conanfile.py: Running CMake.configure()
conanfile.py: RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/Users/Todiq/AppData/Local/Temp" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:/Users/Todiq/AppData/Local/Temp"
-- Using Conan toolchain: C:/Users/Todiq/AppData/Local/Temp/build/generators/conan_toolchain.cmake
-- Conan toolchain: CMAKE_GENERATOR_TOOLSET=v143
-- Conan toolchain: Setting CMAKE_MSVC_RUNTIME_LIBRARY=$<$<CONFIG:Release>:MultiThreadedDLL>
-- Conan toolchain: C++ Standard 14 with extensions OFF
-- Conan toolchain: Including CMakeDeps generated conan_cmakedeps_paths.cmake
-- The CXX compiler identification is MSVC 19.44.35207.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.44.35207/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Configuring Targets for app/1.0
-- Conan: Configuring Targets for libb/1.0
-- Conan: Configuring Targets for liba/1.0
-- Conan: Target declared imported SHARED library 'liba::liba'
-- Conan: Target declared imported SHARED library 'libb::libb'
-- Conan: Target declared imported executable 'app::app' host
-- CONAN_RUNTIME_LIB_DIRS=$<$<CONFIG:Release>:C:/Users/Todiq/AppData/Local/Temp/build/generators/../../app/build/Release>;$<$<CONFIG:Release>:C:/Users/Todiq/AppData/Local/Temp/build/generators/../../conan/p/b/libb2c8ce5759a93c/p/bin>;$<$<CONFIG:Release>:C:/Users/Todiq/AppData/Local/Temp/build/generators/../../conan/p/b/liba8a5b8f46731a9/p/bin>
-- Configuring done (3.8s)
-- Generating done (0.0s)
-- Build files have been written to: C:/Users/Todiq/AppData/Local/Temp/build

conanfile.py: Running CMake.build()
conanfile.py: RUN: cmake --build "C:\Users\Todiq\AppData\Local\Temp\build" --config Release
Version MSBuild 17.14.8+a7a4d5af0 pour .NET Framework

C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(548,5): warning MSB8
029: Le répertoire intermédiaire ou le répertoire de sortie ne peut pas se trouver sous le répertoire temporaire car cela risque de
créer des problèmes avec la génération incrémentielle. [C:\Users\Todiq\AppData\Local\Temp\build\ZERO_CHECK.vcxproj]
  1>Checking Build System
C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(548,5): warning MSB8
029: Le répertoire intermédiaire ou le répertoire de sortie ne peut pas se trouver sous le répertoire temporaire car cela risque de
créer des problèmes avec la génération incrémentielle. [C:\Users\Todiq\AppData\Local\Temp\build\ALL_BUILD.vcxproj]
  Building Custom Rule C:/Users/Todiq/AppData/Local/Temp/CMakeLists.txt

conanfile.py: Running CMake.install()
conanfile.py: RUN: cmake --install "C:\Users\Todiq\AppData\Local\Temp\build" --config Release --prefix "C:/Users/Todiq/AppData/Local/Temp"
CMake Error at cmake_install.cmake:39 (file):
  file INSTALL cannot find
  "C:/Users/Todiq/AppData/Local/Temp/build/generators/../../app/bin/app.exe":
  File exists.



ERROR: conanfile.py: Error in build() method, line 17
        cmake.install()
        ConanException: Error 1 while executing

(venv) C:\Users\Todiq\AppData\Local\Temp>
```

</details>


I tried editing the `layout()` method:

```python
	def layout(self):
		bt = "." if self.settings.get_safe("os") != "Windows" else str(self.settings.build_type)
		ext = "" if self.settings.get_safe("os") != "Windows" else ".exe"
		cmake_layout(self)
		self.cpp.build.exe = "app"
		self.cpp.build.location = os.path.join(bt, f"app{ext}")
		self.cpp.package.exe = "app"
		self.cpp.package.location = os.path.join("bin", f"app{ext}")
```

But I get the same error:
```cmd
CMake Error at cmake_install.cmake:39 (file):
  file INSTALL cannot find
  "C:/Users/Todiq/AppData/Local/Temp/build/generators/../../app/bin/app.exe":
  No error.



ERROR: conanfile.py: Error in build() method, line 17
        cmake.install()
        ConanException: Error 1 while executing
```

--- a/conan/internal/model/cpp_info.py
+++ b/conan/internal/model/cpp_info.py
@@ -15,6 +15,7 @@ _DIRS_VAR_NAMES = ["_includedirs", "_srcdirs", "_libdirs", "_resdirs", "_bindirs
 _FIELD_VAR_NAMES = ["_system_libs", "_package_framework", "_frameworks", "_libs", "_defines",
                     "_cflags", "_cxxflags", "_sharedlinkflags", "_exelinkflags", "_sources"]
 _ALL_NAMES = _DIRS_VAR_NAMES + _FIELD_VAR_NAMES
+_SINGLE_VALUE_VARS = "type", "exe", "location", "link_location", "languages", "package_framework"
 
 
 class MockInfoProperty:
@@ -460,6 +461,12 @@ class _Component:
                     merge_list(other_values, current_values)
                 else:
                     setattr(self, varname, other_values)
+
+        for varname in _SINGLE_VALUE_VARS:  # To allow editable of .exe/.location
+            other_values = getattr(other, varname)
+            if other_values is not None:
+                # Just overwrite the existing value, not possible to append
+                setattr(self, varname, other_values)
 
         if other.requires:
             current_values = self.get_init("requires", [])

--- a/conan/internal/model/cpp_info.py
+++ b/conan/internal/model/cpp_info.py
@@ -15,7 +15,7 @@ _DIRS_VAR_NAMES = ["_includedirs", "_srcdirs", "_libdirs", "_resdirs", "_bindirs
 _FIELD_VAR_NAMES = ["_system_libs", "_package_framework", "_frameworks", "_libs", "_defines",
                     "_cflags", "_cxxflags", "_sharedlinkflags", "_exelinkflags", "_sources"]
 _ALL_NAMES = _DIRS_VAR_NAMES + _FIELD_VAR_NAMES
-_SINGLE_VALUE_VARS = "type", "exe", "location", "link_location", "languages", "package_framework"
+_SINGLE_VALUE_VARS = "_type", "_exe", "_location", "_link_location", "_languages"
 
 
 class MockInfoProperty:

--- a/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
+++ b/test/functional/toolchains/cmake/cmakedeps2/test_cmakeconfigdeps_new.py
@@ -15,12 +15,12 @@ new_value = "will_break_next"
 
 @pytest.mark.tool("cmake")
 class TestExes:
-    @pytest.mark.parametrize("tool_requires", [False, True])
-    def test_exe(self, tool_requires):
+    @pytest.mark.parametrize("editable", [False, True])
+    def test_exe(self, editable):
         conanfile = textwrap.dedent(r"""
-            import os
+            import os, platform
             from conan import ConanFile
-            from conan.tools.cmake import CMake
+            from conan.tools.cmake import CMake, cmake_layout
 
             class Test(ConanFile):
                 name = "mytool"
@@ -29,6 +29,13 @@ class TestExes:
                 settings = "os", "arch", "compiler", "build_type"
                 generators = "CMakeToolchain"
                 exports_sources = "*"
+
+                def layout(self):
+                    cmake_layout(self)
+                    self.cpp.build.exe = "mytool"
+                    name = "mytool.exe" if platform.system() == "Windows" else "mytool"
+                    app_loc = os.path.join("build", str(self.settings.build_type), name)
+                    self.cpp.build.location = app_loc
 
                 def build(self):
                     cmake = CMake(self)
@@ -57,46 +64,48 @@ class TestExes:
         c.run("new cmake_exe -d name=mytool -d version=0.1")
         c.save({"conanfile.py": conanfile,
                 "src/main.cpp": main})
+        if editable:
+            c.run("editable add .")
         c.run("create .")
 
-        requires = "tool_requires" if tool_requires else "requires"
-        consumer = textwrap.dedent(f"""
-            from conan import ConanFile
-            from conan.tools.cmake import CMakeDeps, CMakeToolchain, CMake, cmake_layout
-            class Consumer(ConanFile):
-                settings = "os", "compiler", "arch", "build_type"
-                {requires} = "mytool/0.1"
+        for requires in ("tool_requires", "requires"):
+            consumer = textwrap.dedent(f"""
+                from conan import ConanFile
+                from conan.tools.cmake import CMakeDeps, CMakeToolchain, CMake, cmake_layout
+                class Consumer(ConanFile):
+                    settings = "os", "compiler", "arch", "build_type"
+                    {requires} = "mytool/0.1"
 
-                def generate(self):
-                    deps = CMakeDeps(self)
-                    deps.generate()
-                    tc = CMakeToolchain(self)
-                    tc.generate()
+                    def generate(self):
+                        deps = CMakeDeps(self)
+                        deps.generate()
+                        tc = CMakeToolchain(self)
+                        tc.generate()
 
-                def layout(self):
-                    cmake_layout(self)
-                def build(self):
-                    cmake = CMake(self)
-                    cmake.configure()
-                    cmake.build()
-            """)
-        cmake = textwrap.dedent("""
-            set(CMAKE_C_COMPILER_WORKS 1)
-            set(CMAKE_C_ABI_COMPILED 1)
-            cmake_minimum_required(VERSION 3.15)
-            project(consumer C)
+                    def layout(self):
+                        cmake_layout(self)
+                    def build(self):
+                        cmake = CMake(self)
+                        cmake.configure()
+                        cmake.build()
+                """)
+            cmake = textwrap.dedent("""
+                set(CMAKE_C_COMPILER_WORKS 1)
+                set(CMAKE_C_ABI_COMPILED 1)
+                cmake_minimum_required(VERSION 3.15)
+                project(consumer C)
 
-            find_package(mytool)
-            add_custom_command(OUTPUT out.c COMMAND MyTool::myexe)
-            add_library(myLib out.c)
-            """)
-        c.save({"conanfile.py": consumer,
-                "CMakeLists.txt": cmake}, clean_first=True)
-        c.run(f"build . -c tools.cmake.cmakedeps:new={new_value}")
-        assert "find_package(mytool)" in c.out
-        assert "target_link_libraries(..." not in c.out
-        assert "Conan: Target declared imported executable 'MyTool::myexe'" in c.out
-        assert "Mytool generating out.c!!!!!" in c.out
+                find_package(mytool)
+                add_custom_command(OUTPUT out.c COMMAND MyTool::myexe)
+                add_library(myLib out.c)
+                """)
+            c.save({f"consumer_{requires}/conanfile.py": consumer,
+                    f"consumer_{requires}/CMakeLists.txt": cmake})
+            c.run(f"build consumer_{requires} -c tools.cmake.cmakedeps:new={new_value}")
+            assert "find_package(mytool)" in c.out
+            assert "target_link_libraries(..." not in c.out
+            assert "Conan: Target declared imported executable 'MyTool::myexe'" in c.out
+            assert "Mytool generating out.c!!!!!" in c.out
 
     def test_exe_components(self):
         conanfile = textwrap.dedent(r"""

